### PR TITLE
feat: add --project option to limit what projects are running

### LINKED ISSUE #4519: Allow running specific project using workspaces
### Clear and concise description of the problem

We have a monorepo with several packages. Some of those packages have both integration- and unit-tests in them. We have different vitest.config.*.ts files for them but we currently don't have a way to specify which project to run in the workspace. Our current workaround is to simply run vitest inside the package with the configuration file specified with --configuration.

### Suggested solution

Allow a --project flag on the cli that will execute only the project with that name from the list of projects in the roots vitest.config.ts.

### Alternative

_No response_

### Additional context

_No response_

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide/).
- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by sheremet-va**
### Description

Fixes #4519

### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. If the feature is substantial or introduces breaking changes without a discussion, PR might be closed.
- [ ] Ideally, include a test that fails without this PR but passes with it.
- [ ] Please, don't make changes to `pnpm-lock.yaml` unless you introduce a new test example.

### Tests
- [ ] Run the tests with `pnpm test:ci`.

### Documentation
- [ ] If you introduce new functionality, document it. You can run documentation with `pnpm run docs` command.

### Changesets
- [ ] Changes in changelog are generated from PR name. Please, make sure that it explains your changes in an understandable manner. Please, prefix changeset messages with `feat:`, `fix:`, `perf:`, `docs:`, or `chore:`.

### CODE REVIEW

File: packages/vitest/src/node/core.ts [hunk 1]
```diff
@@ -152,6 +152,9 @@ export class Vitest {
     await Promise.all(this._onSetServer.map(fn => fn()))
 
     this.projects = await this.resolveWorkspace(cliOptions)
+    const filteredProjects = toArray(resolved.project)
+    if (filteredProjects.length)
+      this.projects = this.projects.filter(p => filteredProjects.includes(p.getName()))
```
Comment (AriPerkkio) [inline packages/vitest/src/node/core.ts:157]: Should this be configurable in watch-mode like other patterns are? Might be useful. 

https://github.com/vitest-dev/vitest/blob/d1e1bc90a9685ee153c17b12ab3d1c99d16d9750/packages/vitest/src/node/stdin.ts#L91-L114

File: packages/vitest/src/node/core.ts [hunk 2]
```diff
@@ -512,6 +518,17 @@ export class Vitest {
...
 
+  async changeProjectName(pattern: string) {
+    if (pattern === '')
+      delete this.configOverride.project
+    else
+      this.configOverride.project = pattern
+
+    this.projects = this.resolvedProjects.filter(p => p.getName() === pattern)
+    const files = (await this.globTestFiles()).map(([, file]) => file)
+    await this.rerunFiles(files, `change project filter`)
```
Comment (AriPerkkio) [inline packages/vitest/src/node/core.ts:529]:
```suggestion
    await this.rerunFiles(files, 'change project filter')
```

**Review by AriPerkkio — APPROVED** [review]
Really nice - this is something I've wanted for long time but haven't had time to implement myself. 🙌 

I did some manual testing in run and watch modes. Works well.

File: docs/guide/workspace.md [hunk 3]
```diff
@@ -99,6 +99,49 @@ export default defineProject({
...
+```
+```bash [pnpm]
+pnpm run test
+```
+```bash [bun]
+bun test
+```
+:::
+
+If you need to run tests only inside a single project, use `--project` CLI option:
```
Comment (dammy001) [inline docs/guide/workspace.md:131]:
```suggestion
If you need to run tests only inside a single project, use the `--project` CLI option:
```

**Review by dammy001 — APPROVED** [review]

File: packages/vitest/src/node/logger.ts [hunk 4]
```diff
@@ -95,6 +96,9 @@ export class Logger {
     const comma = c.dim(', ')
     if (filters?.length)
       this.console.error(c.dim('filter:  ') + c.yellow(filters.join(comma)))
+    const projectsFilter = toArray(config.project)
+    if (!projectsFilter.length)
```
Comment (sheremet-va) [inline packages/vitest/src/node/logger.ts:100]:
```suggestion
    if (projectsFilter.length)
```

**Comment by Lewenhaupt** [comment]
Awesome! 👌

**Pull request merged**